Base/QTApp/qSlicerMainWindow.cxx

```diff
@@ -642,9 +642,9 @@ bool qSlicerMainWindowPrivate::confirmCloseApplication()
     messageBox->exec();
     if (messageBox->clickedButton() == saveButton)
       {
-      // \todo Check if the save data dialog was "applied" and close the
-      // app in that case
+      // Save data and close the app
       this->FileSaveSceneAction->trigger();
```

Comment:
This original comment with the `\todo` indicates that the author was concerned about the case where the save failed or was canceled, in which case you wouldn't want the application to quit since it could mean losing data.  Unless that case is accounted for I don't think we should close the app just because the save dialog was triggered.

Reply:
Understood, I will implement this check in that case